src/python/pants/backend/python/util_rules/pex_environment.py

```diff
@@ -141,7 +142,33 @@ async def find_pex_python(
             Get(
                 BinaryPaths,
                 BinaryPathRequest(
-                    search_path=python_setup.interpreter_search_paths, binary_name=binary_name
+                    search_path=python_setup.interpreter_search_paths,
+                    binary_name=binary_name,
+                    test_args=[
```

Comment:
Maybe explain why we use this more complex script, rather than something simple like running the program `shasum`. In particular, explain the Pyenv issue.

Reply:
I don't think re-explaining not using a host binary (`shasum`) is wise to sprinkle about everywhere we don't do it. It should be well understood that we should be using 0 host binaries in Pants since they are not gauranteed to be there, may not be compatible or may not be what you think they are or behave the way you think they behave.

I can add a note though about the pyenv case to ensure we don't remove the hashing code once automatic hashing of the binary path is added in #10769.

Reply:
Sounds good! Thanks.